src/Common/src/System/Runtime/InteropServices/StringBuffer.cs

```diff
@@ -0,0 +1,387 @@
+            }
+        }
+
+        /// <summary>
+        /// Split the contents into strings via the given split characters.
+        /// </summary>
+        /// <exception cref="OverflowException">Thrown if the substring is too big to fit in a string.</exception>
+        unsafe public IEnumerable<string> Split(char splitCharacter)
+        {
+            var strings = new List<string>();
+            char* start = CharPointer;
+            char* current = start;
+
+            ulong length = _length;
+
+            for (ulong i = 0; i < length; i++)
+            {
+                if (splitCharacter == *current)
+                {
+                    // Split
+                    strings.Add(new string(value: start, startIndex: 0, length: checked((int)(current - start))));
+                    start = current + 1;
+                }
+
+                current++;
+            }
+
+            strings.Add(new string(value: start, startIndex: 0, length: checked((int)(current - start))));
+
+            return strings;
+        }
+
+        /// <summary>
+        /// True if the buffer contains the given character.
+        /// </summary>
+        public unsafe bool Contains(char value)
+        {
+            char* start = CharPointer;
+            ulong length = _length;
+
+            for (ulong i = 0; i < length; i++)
+            {
+                if (*start++ == value) return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// True if the buffer contains any of the specified characters.
+        /// </summary>
+        public unsafe bool Contains(params char[] values)
+        {
+            if (values == null || values.Length == 0) return false;
+
+            char* start = CharPointer;
+            ulong length = _length;
+
+            for (ulong i = 0; i < length; i++)
+            {
+                if (ContainsChar(values, *start)) return true;
+                start++;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Trim the specified values from the end of the buffer.
+        /// </summary>
+        public unsafe void TrimEnd(params char[] values)
+        {
+            if (values == null || values.Length == 0 || _length == 0) return;
+
+            char* end = CharPointer + _length - 1;
+
+            while (ContainsChar(values, *end))
+            {
+                Length--;
+                end--;
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool ContainsChar(char[] source, char value)
+        {
+            for (int i = 0; i < source.Length; i++)
+            {
+                if (value == source[i]) return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// String representation of the entire buffer. If the buffer is larger than the maximum size string (int.MaxValue) will truncate.
```

Comment:
How about just throwing if > int.MaxValue, i.e. just use checked((int)_length). I'd rather avoid silent different behavior in an exceptionally rare code path.